plugin/builtin.go

```diff
@@ -1,10 +1,19 @@
 package plugin
+import (
+	"encoding/json"
+)
 
 type getManifest[T any] struct{}
 
 func (instance *getManifest[T]) Call(plugin *Plugin[T], request map[string]any) (map[string]any, error) {
 	result := make(map[string]any)
 	// TODO: feel the getmanifest result
+	result["options"], _ = json.Marshal(plugin.Options)
```

Comment:
```suggestion
	result["options"], _ = json.Marshal(plugin.Options)
```

return a sequence of bytes `json.Marshal` I think with go you can put the option in the array, like `result["options"] = plugin.Options`

The important stuff is that when the `json.Marshall()` is called the string produced about the following json string

```json
{
  "options": [
    {
      "name": "greeting",
      "type": "string",
      "default": "World",
      "description": "What name should I call you?",
      "deprecated": false
    }
  ],
  "rpcmethods": [
    {
      "name": "hello",
      "usage": "[name]",
      "description": "Returns a personalized greeting for {greeting} (set via options)."
    },
    {
      "name": "gettime",
      "usage": "",
      "description": "Returns the current time in {timezone}",
      "long_description": "Returns the current time in the timezone that is given as the only parameter.\nThis description may be quite long and is allowed to span multiple lines.",
      "deprecated": false
    }
  ],
  "subscriptions": [
    "connect",
    "disconnect"
  ],
  "hooks": [
    { "name": "openchannel", "before": ["another_plugin"] },
    { "name": "htlc_accepted" }
  ],
  "featurebits": {
    "node": "D0000000",
    "channel": "D0000000",
    "init": "0E000000",
    "invoice": "00AD0000"
  },
  "notifications": [
    {
	  "method": "mycustomnotification"
	}
  ],
  "dynamic": true
}
```

Reply:
The main error here is that the option is not a map inside the json but an array, we need to make some helper function to convert a map into an array with the correct format.

Do you want help with this? I can make some common function that we can use around

Reply:
@vincenzopalazzo what do you mean by ```Good job, I think it is a good starting point, the only get manifest does not need to call json.Marshall because we call json.Marshall inside the Start method when we return the RPC method.```

What then will the getManifest then do, I thought it is meant to generate the json or do we create a map(or struct) and then turn to a json during start?

Reply:
the get manifest is not a helper function, but a RPC method!

If you take a look the dart version that @swaptr you noted that in the getmanifest are skipped some special rpcMethod with name `init` and `getmanifest`

https://github.com/dart-lightning/lndart.cln/blob/main/packages/cln_plugin/lib/src/cln_plugin_base.dart#L107-L128

In addition, are you sure that`result["options"], _ = json.Marshal(plugin.Options)` generate the json string that we are looking for?

Reply:
```func generateJsonBody(mapData map[string]any) any {
	v := make([]any, 0, len(mapData))

	for  _, value := range mapData {
		v = append(v, value)
	 }
	jsonData, _ := json.Marshal(v)
	return jsonData
}```

Is the kind of function you're talking about.

Reply:
this is not my function, right?

However, the `getManifest` function is not this kind of library but the plugin RPC method take a JSON input as a request and return a JSON output as a response.

So, no it is not the function that I'm talking about.

P.S: the function is also wrong because the `jsonData` is a `[]byte` always, no needed for `any`